packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -498,20 +503,70 @@ export default class Meeting extends StatelessSparkPlugin {
     this.setCorrelationId(this.id);
     // this.state.join();
     this.spark.meetings.notifyRinging(this, OUTGOING);
-    return MeetingUtil.joinMeeting(this)
+    return MeetingUtil.joinMeeting(this, options)
       .then((response) => {
         this.setLocus(response);
         this.spark.meetings.notifyRinging(this, null, false);
         // this.state.connect();
-        // TODO: different scenarios for meeting , claimed and unclaimed
       })
       .catch((err) => {
         this.spark.meetings.notifyRinging(this, null, false);
+        if (MeetingUtil.isPinOrGuest(err)) {
+          if (MeetingUtil.hasOwner(this.meetingInfo)) {
+            return MeetingUtil.joinMeeting(this, {
+              moderator: MeetingUtil.isOwnerSelf(this.owner, this.userId) ? MODERATOR_TRUE : MODERATOR_FALSE
+            })
+              .then((response) => {
+                this.setLocus(response);
+                // this.state.connect();
+              });
+          }
+          // request host pin or non host for unclaimed PMR
+          this.trigger(EVENT_TRIGGERS.LOBBY_HOST_GUEST, {
+            error: err,
+            message: 'The meeting is locked. This is expected behavior. Call #joinAsHost or #joinAsGuest',
+            joinAsHostOrGuest: true
+          });
+          // request host pin or non host for unclaimed PMR
+          // TODO: deal with Error object
+          // eslint-disable-next-line
```

Comment:
Anybody know how to have the Promise reject with an Error object and be able to read it's props? I had remembered doing it one way, but that did not fly with js, so i can spend some more time reading, but if anybody knows offhand...

Reply:
`reject(new Error('your message here'))`. Also takes an object. If you want to add additional properties you can use a [custom error class](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Custom_Error_Types)

I'm trying to work through the `joinAsHostOrGuest` bits. I think it might be better to be generic and trigger a `meeting:inLobby` event or similar. Then with proper documentation the end user can decide what UI they want to show and call the appropriate methods. An example of this is what if there was eventually another user type that could join a meeting as neither host nor guest, etc.